src/ajax/check_hgvs_dialogue.php

```diff
@@ -0,0 +1,603 @@
+ * Programmer  : Ivo F.A.C. Fokkema <I.F.A.C.Fokkema@LUMC.nl>
+ *
+ *
+ * This file is part of LOVD.
+ *
+ * LOVD is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * LOVD is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LOVD.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *************/
+
+define('ROOT_PATH', '../');
+require ROOT_PATH . 'inc-init.php';
+require ROOT_PATH . 'inc-lib-variants.php';
+header('Content-type: text/javascript; charset=UTF-8');
+
+
+// Retrieving the variant and the transcripts to map to.
+$sVariant     = $_REQUEST['var'];
+$aTranscripts = (empty($_REQUEST['transcripts'])? array() : explode('|', $_REQUEST['transcripts']));
+
+// Retrieving the name of the input field.
+$sFieldName   = $_REQUEST['fieldName'];
+
+
+
+// Reset all values.
+print('
+// Resetting all values.
+if ($(\'#variantForm input[name*="VariantOn"]\').hasClass("acc")) {
+    // If any input in the form is of the class acc(epted), this means
+    //  that these input fields were filled in after full mapping and
+    //  validation of VariantValidator. If then, the script is called
+    //  again, we want to RESET these values, since we do not want to
+    //  risk having incoherent variants in the form simultaneously,
+    //  especially not those we have mapped and validated ourselves.
+    // Resetting the transcript fields.        
+    var oTranscriptFields = $(\'#variantForm input[name$="VariantOnTranscript/DNA"]\');
+    oTranscriptFields.val("").removeClass();
+    oTranscriptFields.siblings("img").attr({src: "gfx/trans.png"}).show();
+    $(\'#variantForm input[name$="VariantOnTranscript/RNA"]\').val("").removeClass();
+    $(\'#variantForm input[name$="VariantOnTranscript/Protein"]\').val("").removeClass();
+    
+    // Resetting the genomic fields.
+    var oGenomicVariants = $(\'#variantForm input[name^="VariantOnGenome/DNA"]\');
+    oGenomicVariants.val("").removeClass();
+    oGenomicVariants.siblings("img").attr({src: "gfx/trans.png"}).show();
+}
+');
+
+
+if (!$sVariant) {
+    // If the variant is empty, we can simply close the script.
+    exit;
+}
+
+
+
+// Preparing variables to allow easy and readable updates of the dialogue.
+
+// Preparing the steps.
+$sStepInitialChecks = 'statusChecks';
+$sStepMapping       = 'statusMapping';
+
+// Preparing the images.
+$sImageNeutral = 'gfx/trans.png';
+$sImagePassed  = 'gfx/check.png';
+$sImageFailed  = 'gfx/cross.png';
+$sImageLoading = 'gfx/lovd_loading.gif';
```

Comment:
-> This is personal preference, I think. Personally I think working with variables instead of strings in these cases is easier to read and adjust. Constants would be a bit annoying since they could then be reached in other scripts and that would be unnecessarily confusing. If you prefer it to be a string, that is fine, too.

Reply:
Let's keep things "standard" and just use strings. Thanks!